waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -972,6 +1009,30 @@
   (get-services [this]
     (get-services this))
 
+  (signal-instance [this service-id instance-id signal-type timeout-ms]
+    (ss/try+
+      (let [service (service-id->service this service-id)
```

Comment:
We should have the `http-404-not-found` code paths for the service and instance not existing (like we do for the shell scheduler).